pkgs/applications/editors/codeblocks/default.nix

```diff
@@ -1,81 +1,91 @@
-{ lib, stdenv, fetchurl, fetchpatch, pkg-config, file, zip, wxGTK31, gtk3
-, contribPlugins ? false, hunspell, gamin, boost, wrapGAppsHook
+{ lib
+, stdenv
+, fetchFromGitHub
+, fetchpatch
+, pkg-config
+, autoconf
+, automake
+, file
+, git
+, libtool
+, wrapGAppsHook
+, zip
+, gtk3
+, wxGTK32
+, contribPlugins ? false
+, boost
+, gamin
+, hunspell
 }:
 
-with lib;
-
 stdenv.mkDerivation rec {
-  name = "${pname}-${lib.optionalString contribPlugins "full-"}${version}";
-  version = "20.03";
   pname = "codeblocks";
+  version = "unstable-2022-11-02";
+  name = "${pname}-${lib.optionalString contribPlugins "full-"}${version}";
 
-  src = fetchurl {
-    url = "mirror://sourceforge/codeblocks/Sources/${version}/codeblocks-${version}.tar.xz";
-    sha256 = "1idaksw1vacmm83krxh5zlb12kad3dkz9ixh70glw1gaibib7vhm";
+  src = fetchFromGitHub {
+    owner = "arnholm";
+    repo = "codeblocks_sfmirror";
+    rev = "4d3041fe5201173effe1bda2c8db274f74ff47b2";
+    leaveDotGit = true;
```

Comment:
Why do we need this? Can we patch instead?

Reply:
Arch Linux was already applying [13 patches](https://github.com/archlinux/svntogit-community/blob/packages/codeblocks/trunk/PKGBUILD), maybe more for darwin...

Reply:
LeaveDotGit will result in an unreproducible hash